test/e2e/aks.go

```diff
@@ -87,6 +87,12 @@ func DiscoverAndWaitForAKSControlPlaneInitialized(ctx context.Context, input Dis
 	})
 	Expect(controlPlane).NotTo(BeNil())
 
+	Logf("Waiting for 'Available' status from Azure Resource Health for control plane %s/%s", controlPlane.Namespace, controlPlane.Name)
+	WaitForControlPlaneResourceHealthAvailable(ctx, WaitForControlPlaneResourceHealthAvailableInput{
```

Comment:
Does this considerably slow down the test or is it manageable?

Reply:
It's considerable. About 15-20 minutes last I tried.

Reply:
I just repro'd this locally and it took 20 minutes.

The cluster in every other way seems fine. I'm unsure if this is something that will be user-impacting (obviously it's E2E-impacting and this additional timeout tolerance is appropriate).

What capi and/or capz CRDs are impacted by this change?

Reply:
Until the status turns green and the AzureManagedControlPlane's "Ready" condition turns true, the associated CAPI Cluster and MachinePools and CAPZ AzureManagedMachinePools will also not be marked ready. I'm not sure if any of that would prevents the cluster from actually being usable though.

Reply:
Delaying those transitions to Ready by 20 minutes almost certainly has some knock-on effects. How about we turn this into a `AzureManagedControlPlane` bool configuration property that defaults to false? That way we can expose it to folks who definitely want it, and understand the tradeoffs.

Reply:
I agree, that seems like a good compromise.